### PR TITLE
Show access keys for azure_rm_storageaccount

### PR DESCRIPTION
**Pull request by mmyyrroonn**
##### SUMMARY

Ref: https://stackoverflow.com/questions/43027457/how-to-fetch-azure-rm-storage-account-keys-in-ansible

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

azure_rm_storageaccount
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_storageaccount_facts.py [hunk 1]
```diff
@@ -409,13 +418,17 @@ def account_obj_to_dict(self, account_obj, blob_service_props=None):
                 queue=self.format_endpoint_dict(account_dict['name'], account_key[0], account_obj.primary_endpoints.queue, 'queue'),
                 table=self.format_endpoint_dict(account_dict['name'], account_key[0], account_obj.primary_endpoints.table, 'table')
             )
+            if account_key[0]:
```
Comment (yuwzho) [inline lib/ansible/modules/cloud/azure/azure_rm_storageaccount_facts.py:421]: Move to the L416's dict construct is OK, no need another conditional check

Comment (mmyyrroonn) [reply lib/ansible/modules/cloud/azure/azure_rm_storageaccount_facts.py:421]: If we move it into the dict construct, there will be an empty field in the results like `key: ""` if `I(show_connection_string=false)`. There is a check in the function `self.format_endpoint_dict` to check whether the account_key is empty or not. If it is empty, the results will not contain the `connection_string`. So, is it ok to move it into the dict construction?

Comment (yuwzho) [reply lib/ansible/modules/cloud/azure/azure_rm_storageaccount_facts.py:421]: IMO, we can move the `get_connection_string` and construct the endpoints to one conditional check `if self.show_connection_string`.

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
shipit

**Pull request merged**